native/jsonrs/Cargo.toml

```diff
@@ -10,6 +10,7 @@ path = "src/lib.rs"
 crate-type = ["dylib"]
 
 [dependencies]
+flate2 = { version = "1.0", features = ["zlib-ng"] }
```

Comment:
What's the reason for using zlib-ng here?

Reply:
The flate2 library has good documentation here: https://github.com/rust-lang/flate2-rs#backends

When I did my benchmarking locally, I was getting better results from `zlib-ng` rather than re-using the system `zlib`

Reply:
I think the library default is to use the system zlib to reduce dependencies and to reduce the compiled size of the artifacts, but it made sense to me that our use-case is specifically around high-performance, so I thought it would be best to select the highest performance backend available.

Reply:
According to those flate2 docs, the default is to use a rust implementation, which I'm more comfortable with. Both to avoid complications in the build system and to avoid potential crashes from unsafe zlib C code that would take down the entire BEAM we're running in if it were to segfault or something. Do you remember off hand how significant the performance difference was?

Reply:
I thought I had posted comments about it, but I don't see them in my notes or this PR. I'll go re-run a few benchmarks and get back to you

Reply:
I also ran the benchmark for zlib-ng with compression level set to 7 and got back these results which makes the performance difference between native rust and this library look less dramatic.

* compression is with 1% of native rust
* zlib-ng (level7) vs native rust
  * 1mb sample: native rust is 30.4% slower
  * 10mb sample: native rust is 6.0% slower
  * 100mb sample: native rust is 41.3% slower


<details>
  <summary>Benchmark output for zlib-ng (level 7)</summary>

```
100m compressed to 31265986 bytes
10m compressed to 63889 bytes
1m compressed to 303189 bytes
Operating System: macOS
CPU Information: Apple M1
Number of Available Cores: 8
Available memory: 16 GB
Elixir 1.14.3
Erlang 25.2.3

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 1 min
memory time: 3 s
reduction time: 0 ns
parallel: 1
inputs: 100m, 10m, 1m
Estimated total run time: 3.25 min

Benchmarking encode_gzip with input 100m ...
Benchmarking encode_gzip with input 10m ...
Benchmarking encode_gzip with input 1m ...

##### With input 100m #####
Name                  ips        average  deviation         median         99th %
encode_gzip          0.52         1.91 s     ±1.90%         1.89 s         2.01 s

Memory usage statistics:

Name           Memory usage
encode_gzip           184 B

**All measurements for memory usage were the same**

##### With input 10m #####
Name                  ips        average  deviation         median         99th %
encode_gzip         49.67       20.13 ms    ±15.68%       19.64 ms       34.02 ms

Memory usage statistics:

Name           Memory usage
encode_gzip           184 B

**All measurements for memory usage were the same**

##### With input 1m #####
Name                  ips        average  deviation         median         99th %
encode_gzip         49.10       20.37 ms     ±9.03%       20.20 ms       22.65 ms

Memory usage statistics:

Name           Memory usage
encode_gzip           184 B

**All measurements for memory usage were the same**
```
</details>

Reply:
With that zlib-ng (level 7) comparison, I tend to lean towards the native rust version. We'll be keeping with a safer implementation and getting most of the performance with less dependencies. What do you think?